apache/incubator-druid
# Druid Doctor

![image](https://user-images.githubusercontent.com/177816/66796429-7a9eca80-eebc-11e9-8c45-f90a6d817c53.png)

Adding a testing and troubleshooting framework to the console.

The Druid Doctor is a dialog that runs a series of tests that test the capabilities of a cluster via the HTTP APIs available via the router node. The aim is to be able to identify common cluster and data misconfigurations and present helpful suggestions to the user.

There are currently 9 tests:

- Verify own status
- Verify own runtime properties
- Verify the Coordinator and Overlord status
- Verify the Coordinator and Overlord runtime properties
- Verify that the sampler works
- Verify that SQL works
- Verify that there are historical nodes
- Verify that the historicals are not overfilled
- Look for time chunks that could benefit from compaction

In the future I would also like to add a test for task slots and pending tasks
